framework/include/base/Attributes.h

```diff
@@ -285,19 +285,30 @@ class AttribPreIC : public Attribute
 class AttribPreAux : public Attribute
 {
 public:
-  typedef bool Key;
-  void setFrom(Key k) { _val = k; }
+  typedef unsigned int Key;
+  void setFrom(Key k)
+  {
+    _vals.clear();
+    _vals.push_back(k);
+  }
 
   AttribPreAux(TheWarehouse & w) : Attribute(w, "pre_aux") {}
-  AttribPreAux(TheWarehouse & w, bool pre_aux) : Attribute(w, "pre_aux"), _val(pre_aux) {}
+  AttribPreAux(TheWarehouse & w, unsigned int val) : Attribute(w, "pre_aux")
+  {
+    _vals.push_back(val);
+  }
+  AttribPreAux(TheWarehouse & w, const std::vector<unsigned int> & vals)
+    : Attribute(w, "pre_aux"), _vals(vals)
+  {
+  }
   virtual void initFrom(const MooseObject * obj) override;
   virtual bool isMatch(const Attribute & other) const override;
   virtual bool isEqual(const Attribute & other) const override;
-  hashfunc(_val);
+  hashfunc(_vals);
   clonefunc(AttribPreAux);
 
 private:
-  bool _val = false;
+  std::vector<unsigned int> _vals;
```

Comment:
I don't think this should be a vector.  Rather than accumulating state, you just overwrite states progressively.

Reply:
I think these attrib classes are somewhat confusing, including things like this data member being a vector. But I would prefer to leave `AttribPreAux` as similar to `AttribExecOns` as possible because they now behave similarly.

As noted in a separate comment, the only difference between the two classes now is how the _vals object is filled.

Reply:
I still want `AttribPreAux` and `AttribPostAux` to be similar to `AttribExecOns`, but if you have a strong opinion about this data structure, I am will to change the vector to a set